pkg/kubectl/resource/builder.go

```diff
@@ -539,17 +547,21 @@ func (b *Builder) visitorResult() *Result {
 		if err != nil {
 			return &Result{err: err}
 		}
-
 		visitors := []Visitor{}
 		for _, mapping := range mappings {
 			client, err := b.mapper.ClientForMapping(mapping)
 			if err != nil {
 				return &Result{err: err}
 			}
+
 			selectorNamespace := b.namespace
 			if mapping.Scope.Name() != meta.RESTScopeNameNamespace {
 				selectorNamespace = ""
 			}
+			if err := b.validateNamespace(client, selectorNamespace); err != nil {
```

Comment:
Is this being run in all cases?  I'm not super-familiar with the code, but it looks like this is in the primary path, not just the error path.